### PR TITLE
Add flask-login "next" parameter support

### LINKED ISSUE #16: follow flask_login "next" parameter on login
When using flask-login, we have the capability of using a "next" get parameter on the login route, which provides the path we should redirect to after a successful login. If the parameter is not provided, the after-login route may be used.

Example, user wants to get to `/protect-all-the-things`. They should get redirected to `/login?next=protect-all-the-things`. Once logged in, get them back to `/protect-all-the-things`.

### PR DESCRIPTION
**Pull request by tjlevel12**
Fixes #16

### CODE REVIEW

File: keg_auth/views.py [hunk 1]
```diff
@@ -64,6 +65,17 @@ def assign_template_vars(self, form):
         self.assign('page_title', self.page_title)
         self.assign('page_heading', self.page_heading)
 
+    @staticmethod
+    def is_safe_url(target):
```
[the remark this answers is not in the document]
Comment (tjlevel12) [reply keg_auth/views.py]: Possibly. I simply based this on the [Flask-Login page](https://flask-login.readthedocs.io/en/latest/#login-example), which says:
> Warning: You MUST validate the value of the next parameter. If you do not, your application will be vulnerable to open redirects.

It then gives a link to the above [`is_safe_url` snippet](http://flask.pocoo.org/snippets/62/).

Is there any harm in checking the URL in totality? ie, is there any benefit to only checking for a leading slash?

[the remark this answers is not in the document]
Comment (rsyring) [reply keg_auth/views.py]: To me, it's conceptually simpler to think of it as: We support forwarding to anywhere on our site, not anywhere on the web.  In that case, I only support a path, not a full URL.  I also had to read the code a couple times to figure out what was going on (although that isn't a deal killer).

It looks from the tests like what you are doing works, so I'm not opposed to keeping it this way.

**Comment by guruofgentoo** [comment]
Note: we should support session option as well, where `next` is in the session instead of the URL.

**Comment by guruofgentoo** [comment]
This change has been included in the branch for #12. What's there includes support for the session option.

**Pull request closed without merging**